source/1_engine/rex_engine/src/cmdline.cpp

```diff
@@ -0,0 +1,205 @@
+#include "rex_engine/cmdline.h"
+
+#include "rex_engine/cmd_args.h"
+#include "rex_engine/diagnostics/assert.h"
+#include "rex_engine/diagnostics/logging/log_macros.h"
+#include "rex_engine/log.h"
+#include "rex_std/cstring.h"
+#include "rex_std/vector.h"
+
+namespace rex
+{
+  namespace cmdline
+  {
+    struct ActiveArgument
+    {
+      rsl::string_view argument;
+      rsl::string_view value;
+    };
+
+    class CommandLineArguments
+    {
+    public:
+      explicit CommandLineArguments(rsl::string_view cmdLine)
+      {
+        // verify the auto generated command line arguments
+        REX_ASSERT_X(verify_args(g_command_line_args.data(), g_command_line_args.size()), "You have ambuguous command line arguments");
```

Comment:
Not sure if this should be an assert, we will be running auto tests in release and probably have release versions of the engine as well. there might still be some "ambiguous" arguments in release which would result in issues in release/auto tests. I think and error or a REX_VERIFY_X ( asserts in release ) might be in order here.

Reply:
it doesn't process ambiguous arguments and will always process the first one, that's why this is an assert here.
I don't think it matters if we have a release version of the engine or not, if you have 2 arguments in the list, eg 1 coming from the engine, 1 coming from the client, the engine doesn't know which one you want to use, which is why there's an assert here.

Reply:
Yes but the problem is that this is only caught in debug ... not sure if this scenario will ever occur in debug

Reply:
It can in debug but if you're comment means to also check this in release, that's fine

Reply:
Yes, but I was wondering if there might be more and if so I don't wanna repeat code for checking an assert in release so that is why I proposed a REX_VERIFY

Reply:
you can have a assert in release or something else, but then it's not really an assert (or it doesn't have to be behind a macro anyway)
it's a runtime check that's always there, regardless of the configuration you run